sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3698 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+
+        StepVerifier.create(fc.read())
+            .assertNext(r -> assertEquals(0, r.array().length))
+                .verifyComplete();
+
+    }
+
+    // This is to test the appropriate integration of DownloadResponse, including setting the correct range values on
+    // HttpGetterInfo.
+    @Test
+    public void readWithRetryRange() {
+        // We are going to make a request for some range on a blob. The Flux returned will throw an exception, forcing
+        // a retry per the DownloadRetryOptions. The next request should have the same range header, which was generated
+        // from the count and offset values in HttpGetterInfo that was constructed on the initial call to download. We
+        // don't need to check the data here, but we want to ensure that the correct range is set each time. This will
+        // test the correction of a bug that was found which caused HttpGetterInfo to have an incorrect offset when it
+        // was constructed in BlobClient.download().
+        DataLakeFileAsyncClient fileAsyncClient = getFileAsyncClient(getDataLakeCredential(), fc.getPathUrl(),
+            new MockRetryRangeResponsePolicy("bytes=2-6"));
+
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        // Because the dummy Flux always throws an error. This will also validate that an IllegalArgumentException is
+        // NOT thrown because the types would not match.
+
+        StepVerifier.create(fileAsyncClient.readWithResponse(new FileRange(2, 5L),
+                new DownloadRetryOptions().setMaxRetryRequests(3), null, false))
+            .assertNext(r -> {
+                StepVerifier.create(r.getValue())
+                    .verifyErrorSatisfies(p -> {
+                        assertInstanceOf(IOException.class, p);
+                    });
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void readMin() {
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        StepVerifier.create(FluxUtil.collectBytesInByteBufferStream(fc.read()))
+                .assertNext(r -> TestUtils.assertArraysEqual(DATA.getDefaultBytes(), r))
+                .verifyComplete();
+    }
+
+    @ParameterizedTest
+    @MethodSource("readRangeSupplier")
+    public void readRange(long offset, Long count, String expectedData) {
+        FileRange range = (count == null) ? new FileRange(offset) : new FileRange(offset, count);
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        ByteArrayOutputStream readData = new ByteArrayOutputStream();
+
+        StepVerifier.create(fc.readWithResponse(range, null, null, false))
+            .assertNext(r -> {
+                r.getValue().subscribe(piece -> {
+                    try {
+                        readData.write(piece.array());
```

Comment:
Use `FluxUtil.collectBytesInByteBufferStream` instead of writing to the OutputStream